docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
@ijuma I completely agree. Versioning the the format is important. It acts as valuable documentation and (when used correctly) can help improve compatibility. 

Let me be clear about my intentions. I am making no comment on the future of zookeeper json compatibility, just this change and its impact to the upcoming release. Given that we still, even in this patch, don't support removing fields in future versions. Changing the version for that reason is a moot point. Therefore the only value the version has is documentation. 

If we weigh documentation and compatibility for 0.10 and choose documentation, then we can bump the version. If we think compatibility is the most important thing to maintain. That can be solved by keeping the version at 2.